python/src/test/java/org/apache/zeppelin/python/IPythonInterpreterTest.java

```diff
@@ -235,4 +238,30 @@ public void testGrpcFrameSize() throws InterpreterException, IOException {
     assertEquals(InterpreterResult.Code.SUCCESS, result.code());
   }
 
+  @Test
+  public void testIPythonProcessKilled() throws InterruptedException, TimeoutException {
+    final Waiter waiter = new Waiter();
+    Thread thread = new Thread() {
+      @Override
+      public void run() {
+        try {
+          InterpreterResult result = interpreter.interpret("import time\ntime.sleep(1000)",
+                  getInterpreterContext());
+          waiter.assertEquals(InterpreterResult.Code.ERROR, result.code());
```

Comment:
Junit can only catch assertion fail in main thread, so here use concurrentunit to do unit test in multiple thread